packages/core-components/src/layout/HeaderLabel/HeaderLabel.test.tsx

```diff
@@ -51,4 +51,29 @@ describe('<HeaderLabel />', () => {
     expect(rendered.getByText('Value')).toBeInTheDocument();
     expect(anchor.href).toBe('http://localhost/test');
   });
+
+  it('should use a `p` tag if the provided value is a string', async () => {
+    const rendered = await renderInTestApp(
+      <HeaderLabel label="Label" value="Value" />,
+    );
+    expect(rendered.getByText('Value').tagName).toBe('P');
+  });
+
+  it('should use a `span` tag if the provided value is not a string', async () => {
+    const rendered = await renderInTestApp(
+      <HeaderLabel label="Label" value={<>Value</>} />,
+    );
+    expect(rendered.getByText('Value').tagName).toBe('SPAN');
+  });
+
+  it('should use the correct custom typography root component', async () => {
+    const rendered = await renderInTestApp(
+      <HeaderLabel
+        label="Label"
+        value="Value"
+        contentTypograpyRootComponent="tr"
```

Comment:
Question:

It feels that the component name was meant to be used only with page headers and not in a table as in this test example. Do you feel it would be a moment to rename the component for something like `EntityLabel`?

Reply:
In this case, I'm using a `tr` in this test case specifically because I was using it to test the functionality because there would be no other reason for a `tr` to exist unless it was working correctly.

Reply:
I thought this `HeaderLabel` was created to be used only with page headers, and seeing it being used in entity cards seems like an opportunity to remove the `Header` prefix from the component name. But I don't have enough context about it and this refactor may not be necessary at this time or it can be done in a separate pull request. Let me ask the @backstage/maintainers what they think.

Reply:
Oh yes I understand what you mean. Yes that would make sense to update the name because we these are actually used in non-header contexts like in the about cards.

Reply:
That sounds reasonable to me, less confusion